### PR TITLE
mark 5 tests as no longer failing on net6

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

* mark 4 tests no longer failing
* mark 1 test as no longer failing, mark as platform win, and do small refactoring so it could potentially run on linux when we actually get the tests running there.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

mark net6 tests correctly.

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Tools/DynamoShapeManager/Utilities.cs [hunk 1]
```diff
@@ -309,21 +309,7 @@ public static Version GetInstalledAsmVersion2(IEnumerable<Version> versions, ref
                 }
 
                 // Fallback mechanism, look inside libg folders if any of them contains ASM dlls.
-                foreach (var v in versions)
```
Comment (mjkkirschner) [inline src/Tools/DynamoShapeManager/Utilities.cs:312]: I extracted this fallback path to a helper method.

File: test/DynamoCoreTests/Logging/AnalyticsServiceTest.cs [hunk 2]
```diff
@@ -4,15 +4,18 @@
...
 using NUnit.Framework;
+using System.Runtime.Versioning;
 
 namespace Dynamo.Tests.Loggings
 {
     public class DynamoAnalyticsDisableTest
     {
-        [Test,Category("FailureNET6")]//TODO this test requires finding ASM using the registry, will not run on linux.
+        [Test]
+        [Platform("win")]//nunit attribute for now only run on windows until we know it's useful on linux.
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/Logging/AnalyticsServiceTest.cs:18]: this is the nunit attribute - it exists in both nunit 2 and 3 without changes.

File: test/DynamoCoreTests/Logging/AnalyticsServiceTest.cs [hunk 3]
```diff
@@ -28,8 +31,17 @@ public void DisableAnalytics()
             var locatedPath = string.Empty;
             var coreDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
             Process dynamoCLI = null;
-
-            DynamoShapeManager.Utilities.GetInstalledAsmVersion2(versions, ref locatedPath, coreDirectory);
+            //TODO an approach we could take to get this running on linux.
+            //unclear if this needs to be compiled with an ifdef or runtime is ok.
+            //related to https://jira.autodesk.com/browse/DYN-5705
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/Logging/AnalyticsServiceTest.cs:37]: my intention with this is that when we work on getting the tests running on linux this should be ready to go.

File: test/DynamoCoreTests/Configuration/PreferenceSettingsTests.cs [hunk 4]
```diff
@@ -375,7 +375,7 @@ PreferencesComparison comparePrefenceSettings(PreferenceSettings defaultSettings
         }
 
         [Test]
-        [Category("UnitTests"), Category("FailureNET6")]
+        [Category("UnitTests")]
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/Configuration/PreferenceSettingsTests.cs:378]: unfortunately this is passing locally, but fails on CI - I will investigate further...
```
  The file D:\jenkins\workspace\Dynamo\DynamoSelfServe\pullRequestValidation\Dynamo\test\settings\DynamoSettings-NewSettings.xml exist: True | DiffProps: 1 | TotProps: 60 | Default Sett NumberFormat: f3 | New Sett NumberFormat: f3 | First Property with the same value IsFirstRun
  Expected: True
  But was:  False
```

**Review by pinzart90 — APPROVED** [review]

**Review by aparajit-pratap — APPROVED** [review]

**Comment by mjkkirschner** [comment]
thanks for the review @aparajit-pratap @pinzart90 - I'd really like to get these tests to pass before merging... build machine troubles it seems.

**Pull request merged**